### PR TITLE
fix(sdk): fix permissions error

### LINKED ISSUE #3297: bucket.get AccessDenied in test
### I tried this:

```js
bring cloud;
bring util;

let bucket = new cloud.Bucket();
let queue = new cloud.Queue();

queue.setConsumer(inflight (message: str) => {
  bucket.put("wing.txt", "Hello, ${message}");
}, timeout: 1s);

test "Hello, world!" {
  let var timeout = 1s;
  if util.env("WING_TARGET") != "sim" {
    timeout = 10s;
  }

  queue.push("world!");
  util.sleep(timeout);
  assert("Hello, world!" == bucket.get("wing.txt"));
}
```

[playground](https://play.winglang.io/?code=YgByAGkAbgBnACAAYwBsAG8AdQBkADsACgBiAHIAaQBuAGcAIAB1AHQAaQBsADsACgAKAGwAZQB0ACAAYgB1AGMAawBlAHQAIAA9ACAAbgBlAHcAIABjAGwAbwB1AGQALgBCAHUAYwBrAGUAdAAoACkAOwAKAGwAZQB0ACAAcQB1AGUAdQBlACAAPQAgAG4AZQB3ACAAYwBsAG8AdQBkAC4AUQB1AGUAdQBlACgAKQA7AAoACgBxAHUAZQB1AGUALgBzAGUAdABDAG8AbgBzAHUAbQBlAHIAKABpAG4AZgBsAGkAZwBoAHQAIAAoAG0AZQBzAHMAYQBnAGUAOgAgAHMAdAByACkAIAA9AD4AIAB7AAoAIAAgAGIAdQBjAGsAZQB0AC4AcAB1AHQAKAAiAHcAaQBuAGcALgB0AHgAdAAiACwAIAAiAEgAZQBsAGwAbwAsACAAJAB7AG0AZQBzAHMAYQBnAGUAfQAiACkAOwAKAH0ALAAgAHQAaQBtAGUAbwB1AHQAOgAgADEAcwApADsACgAKAHQAZQBzAHQAIAAiAEgAZQBsAGwAbwAsACAAdwBvAHIAbABkACEAIgAgAHsACgAgACAAbABlAHQAIAB2AGEAcgAgAHQAaQBtAGUAbwB1AHQAIAA9ACAAMQBzADsACgAgACAAaQBmACAAdQB0AGkAbAAuAGUAbgB2ACgAIgBXAEkATgBHAF8AVABBAFIARwBFAFQAIgApACAAIQA9ACAAIgBzAGkAbQAiACAAewAKACAAIAAgACAAdABpAG0AZQBvAHUAdAAgAD0AIAAxADAAcwA7AAoAIAAgAH0ACgAKACAAIABxAHUAZQB1AGUALgBwAHUAcwBoACgAIgB3AG8AcgBsAGQAIQAiACkAOwAKACAAIAB1AHQAaQBsAC4AcwBsAGUAZQBwACgAdABpAG0AZQBvAHUAdAApADsACgAgACAAYQBzAHMAZQByAHQAKAAiAEgAZQBsAGwAbwAsACAAdwBvAHIAbABkACEAIgAgAD0APQAgAGIAdQBjAGsAZQB0AC4AZwBlAHQAKAAiAHcAaQBuAGcALgB0AHgAdAAiACkAKQA7AAoAfQA%3D)

run 

```
wing test --progress -t tf-aws main.w
```

### This happened:

```
✔ Compiling main.w to tf-aws...
✔ terraform init
✔ terraform apply
✔ Setting up test runner...
✔ Running tests...
fail ┌ main.tfaws » root/Default/env0/test:Hello, world!
     │ Error: Invoke failed with message: "Unhandled". Full error: "{"errorType":"Error","errorMessage":"Object does not exist (key=wing.txt): AccessDenied: Access Denied\n    at throwDefaultError (/var/task/index.js:7589:24)\n    at /var/task/index.js:7599:40\n    at de_GetObjectCommandError (/var/task/index.js:23840:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /var/task/index.js:1503:24\n    at async /var/task/index.js:5180:22\n    at async /var/task/index.js:2895:42\n    at async /var/task/index.js:30832:22\n    at async /var/task/index.js:1747:26\n    at async BucketClient.get (/var/task/index.js:36257:18)","trace":["Error: Object does not exist (key=wing.txt): AccessDenied: Access Denied","    at throwDefaultError (/var/task/index.js:7589:24)","    at /var/task/index.js:7599:40","    at de_GetObjectCommandError (/var/task/index.js:23840:18)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async /var/task/index.js:1503:24","    at async /var/task/index.js:5180:22","    at async /var/task/index.js:2895:42","    at async /var/task/index.js:30832:22","    at async /var/task/index.js:1747:26","    at async BucketClient.get (/var/task/index.js:36257:18)","    at BucketClient.get (/var/task/index.js:36259:17)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async $Closure2.handle (/var/task/index.js:55:36)","    at async exports.handler (/var/task/index.js:44167:10)"]}"
     │     at FunctionClient.invoke (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/shared-aws/function.inflight.js:21:19)
     │     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
     │     at async TestRunnerClient.runTest (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/shared-aws/test-runner.inflight.js:27:13)
     │     at async /Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/commands/test.js:294:30
     │     at async withSpinner (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/util.js:35:24)
     │     at async testTfAws (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/commands/test.js:291:25)
     │     at async testOne (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/commands/test.js:125:20)
     │     at async test (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/commands/test.js:55:39)
     └     at async Command.<anonymous> (/Users/sebastian/.nvm/versions/node/v18.14.2/lib/node_modules/winglang/dist/cli.js:17:30)
```

### I expected this:

No error and correct permissions

### Is there a workaround?

_No response_

### Component

Compiler

### Wing Version

0.23.26

### Wing Console Version

_No response_

### Node.js Version

v18.14.2

### Platform(s)

MacOS

### Anything else?

Looks a lot similar to https://github.com/winglang/wing/issues/3112

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by tsuf239**
fixes #3297

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://www.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/shared-aws/permissions.ts [hunk 1]
```diff
@@ -95,7 +95,10 @@ export function calculateBucketPermissions(
     ops.includes(cloud.BucketInflightMethods.EXISTS) ||
     ops.includes(cloud.BucketInflightMethods.TRY_GET) ||
     ops.includes(cloud.BucketInflightMethods.TRY_GET_JSON) ||
-    ops.includes(cloud.BucketInflightMethods.TRY_DELETE)
+    ops.includes(cloud.BucketInflightMethods.TRY_DELETE) ||
+    // get requires list permissions too https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/GetObjectCommand/
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/shared-aws/permissions.ts:99]: According to the docs you linked, having List permissions only changes the error you get  when an object isn't found. So it's not actually needed. It's possible I'm misreading it though.
 
Did you config that this change fixes the original issue?

**Comment by tsuf239** [comment]
No you’re right,

If you don’t have the s3:ListBucketpermission, Amazon S3 returns an HTTP
status code 403 ("access denied") error.

In the example code Sebastian sent the key didn’t exist, and that’s why he
got that message, it’s still regression though because we were usually
sending the other error

בתאריך יום ה׳, 6 ביולי 2023 ב-23:37 מאת Mark McCulloh <
***@***.***>:

**Review by MarkMcCulloh — APPROVED** [review]
> In the example code Sebastian sent the key didn’t exist, and that’s why he
got that message

Unless I'm misreading, the key should exist though. "wing.txt" should be added to the bucket after `queue.push("world!")` and a 10s sleep in aws. I figured that was the bug, rather than anything wrong with his code (copied below for reference)

```ts
bring cloud;
bring util;

let bucket = new cloud.Bucket();
let queue = new cloud.Queue();

queue.setConsumer(inflight (message: str) => {
  bucket.put("wing.txt", "Hello, ${message}");
}, timeout: 1s);

test "Hello, world!" {
  let var timeout = 1s;
  if util.env("WING_TARGET") != "sim" {
    timeout = 10s;
  }

  queue.push("world!");
  util.sleep(timeout);
  assert("Hello, world!" == bucket.get("wing.txt"));
}
```

> it’s still regression though because we were usually
sending the other error

I do agree with this, so I'll approve the change in this PR but I'm not sure this fixes the linked issue.

File: libs/wingsdk/src/shared-aws/permissions.ts (hunk 1 above)
Comment (MarkMcCulloh) [reply libs/wingsdk/src/shared-aws/permissions.ts:99]: From the original issue:

> I expected this:
> No error and correct permissions

Not just a different (better) error

**Comment by tsuf239** [comment]
> ```ts
> timeout: 1s
> ```

I think it has to do something with the `setConsumer`'s timeout, when I set it to 30 it works :)

**Pull request merged**